src/server/master_server/lock_manager.cc

```diff
@@ -7,70 +7,106 @@
 namespace gfs {
 namespace server {
 
+/* A private helper class to compute the index in the filePathLocks_ collection for a
+ * pathanme. */
+inline uint16_t LockManager::bucket_id(const std::string& pathname) const {
+   return std::hash<std::string>{}(pathname) % shard_size_;
+}
+
 /* Initialize the meta locks and the vector of filePathLocks  */
 LockManager::LockManager() {
   shard_size_ = std::max(std::thread::hardware_concurrency(), (unsigned int) 1);
   metaLocks_ = std::vector<absl::Mutex*>(shard_size_, new absl::Mutex());
   filePathLocks_ =
       std::vector<absl::flat_hash_map<std::string, absl::Mutex*>>(shard_size_);
-  globalLock_ = new absl::Mutex();
 }
 
-bool LockManager::Exist(const std::string& pathname) {
-  auto idx(std::hash<std::string>{}(pathname) % shard_size_);
+bool LockManager::Exist(const std::string& pathname) const {
+  auto idx(bucket_id(pathname));
```

Comment:
Are we trying to essentially use the RAII wrapper property of MutexLock to acquire a metadata lock until the function returns? if so, can we make an explicit comment in at least one of the functions to make this design choice and MutexLock clear to the reader

Reply:
As far as I understand, RAII wrapper for mutex is super common and should be preferred for mutex because of all the early-return conditions. I am not seeing this as a "design choice" per se. I will add a comment for this design though.